proof/crefine/RISCV64/IpcCancel_C.thy

```diff
@@ -2913,12 +2913,120 @@ lemma reply_remove_tcb_ccorres:
     (replyRemoveTCB tptr) (Call reply_remove_tcb_'proc)"
 sorry (* FIXME RT: reply_remove_tcb_corres *)
 
+lemma updateReply_eq:
```

Comment:
Not a great name, `_eq` lemma name would imply some kind of equality between two things... but seeing the `_lemma4` below, it looks like you copied it from somewhere else?

Reply:
Yes, I should have made that more clear when I pushed the commit. All of new lemmas in the `reply_unlink_ccorres` commit, aside from `reply_unlink_ccorres` itself, are derived (in some cases very closely) from analogous lemmas found elsewhere in CRefine.

Reply:
Well, then I guess you're consistent with someone's historic poor naming convention, so it's fine to leave it. Also, apart from the potential optimisation, I'm also fine with your latest commit.

Reply:
`updateReply_eq` is mostly copied from `updateSchedContext_eq`, and `updateReply_ccorres_lemma4` is mostly copied from `updateSchedContext_ccorres_lemma4`. I don't totally understand the `_ccorres_lemma[234]` naming convention, but I (perhaps recklessly) just went with it.

Reply:
I think that's the right move. The whole lemma4 thing has something to do with handling updates of structs within structs at a pointer location, but not sure of the exact history of the 2/3/4. If we ever update it to better tech, having the lemma4's around to grep for will be useful.